pkg/asset/manifests/ingress.go

```diff
@@ -162,6 +162,41 @@ func (ing *Ingress) generateDefaultIngressController(config *types.InstallConfig
 			},
 		}
 		return yaml.Marshal(obj)
+	default:
+		obj := &operatorv1.IngressController{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: operatorv1.GroupVersion.String(),
+				Kind:       "IngressController",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "openshift-ingress-operator",
+				Name:      "default",
+			},
+			Spec: operatorv1.IngressControllerSpec{
+				EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+					Type: operatorv1.LoadBalancerServiceStrategyType,
```

Comment:
It's nuanced, but I think this is a breaking change for all `HostNetworkStrategyType` default platforms (libvirt, OpenStack, ovirt, nutanix, etc...), see [setDefaultPublishingStrategy](https://github.com/openshift/cluster-ingress-operator/blob/19eced8a48126c0b54b5b779ba8169f4c4e4f5a1/pkg/operator/controller/ingress/controller.go#L443-L454).

This logic is now always creating the `default` IngressController and it always sets `spec.endpointPublishingStrategy.type` to `LoadBalancerServiceStrategyType`. For hostnetwork-default platforms, like OpenStack, the `installconfig.publish` field will be left empty, so the `default` IngressController will not be created by the Installer, but it WILL be created in [ensureDefaultIngressController](https://github.com/openshift/cluster-ingress-operator/blob/8b768d2d18d8f889f47e99c72c9a10de6ebcb15a/pkg/operator/operator.go#L452) without `spec.endpointPublishingStrategy.type`, so that it defaults to `HostNetwork` in `setDefaultPublishingStrategy`.

So, we need to be careful not to change the behavior for these non-LoadBalancerService platforms. I think for now you should only create the `default` manifest only when you need to (aka only when you need to specify EIPs).

Reply:
made changes. PTAL.